particles.js

```diff
@@ -1309,42 +1313,50 @@ var pJS = function(tag_id, params){
   };
 
 
-  pJS.fn.vendors.draw = function(){
+  pJS.fn.vendors.draw = function (timestamp) {
 
-    if(pJS.particles.shape.type == 'image'){
-
-      if(pJS.tmp.img_type == 'svg'){
-
-        if(pJS.tmp.count_svg >= pJS.particles.number.value){
-          pJS.fn.particlesDraw();
-          if(!pJS.particles.move.enable) cancelRequestAnimFrame(pJS.fn.drawAnimFrame);
-          else pJS.fn.drawAnimFrame = requestAnimFrame(pJS.fn.vendors.draw);
-        }else{
-          //console.log('still loading...');
-          if(!pJS.tmp.img_error) pJS.fn.drawAnimFrame = requestAnimFrame(pJS.fn.vendors.draw);
-        }
-
-      }else{
+  	//if (pageHidden) {
+  	//	return;
+  	//}
 
-        if(pJS.tmp.img_obj != undefined){
-          pJS.fn.particlesDraw();
-          if(!pJS.particles.move.enable) cancelRequestAnimFrame(pJS.fn.drawAnimFrame);
-          else pJS.fn.drawAnimFrame = requestAnimFrame(pJS.fn.vendors.draw);
-        }else{
-          if(!pJS.tmp.img_error) pJS.fn.drawAnimFrame = requestAnimFrame(pJS.fn.vendors.draw);
-        }
-
-      }
-
-    }else{
-      pJS.fn.particlesDraw();
-      if(!pJS.particles.move.enable) cancelRequestAnimFrame(pJS.fn.drawAnimFrame);
-      else pJS.fn.drawAnimFrame = requestAnimFrame(pJS.fn.vendors.draw);
+    // FPS limit logic
+    // If we are too fast, just draw without updating
+    var fps_limit = pJS.fps_limit;
+    if(timestamp < lastFrameTime + (1000 / fps_limit)) {
+    	requestAnimFrame(pJS.fn.vendors.draw);
```

Comment:
I may be wrong but I think requestAnimationFrame **MUST** be called on each loop but you can then check if the function is not suppose to run and just return.

So basically but the requestAnimFrame(...) before the if statement.

Reply:
In the code below it's called the function, it's handled by other `if/else` because the animation stops if the `particles.move.enable` flag is disabled (and it shouldn't IMO). This`if` is used because it's needed to check the delta between the previous frame to see if a new draw is needed, if it's not needed the values are not updated so the next `rAF` call could be the right one to update everything. For example, if you want 30FPS instead of 60, you need to skip one frame, because `rAF` is called the fastest possible depending on the monitor refresh rate (60 Hz - 60 fps).

The issue solved by this PR is that this library doesn't handle well different frequencies, on my MBP 2021 when I use Chrome I have 120FPS I see faster animations than Safari, limited to 60FPS. Every speed is doubled on Chrome, and this is a bad behavior.